infra/terraform/modules/service/batch.tf

```diff
@@ -0,0 +1,134 @@
+
+module "batch" {
+  source  = "terraform-aws-modules/batch/aws"
+  version = "~> 2.0.0"
+
+  instance_iam_role_name        = "${var.environment}-batch-app-ecs-instance-role"
+  instance_iam_role_path        = "/batch/"
+  instance_iam_role_description = "IAM instance role/profile for AWS Batch ECS instance(s)"
+  instance_iam_role_tags = {
+    ModuleCreatedRole = "Yes"
+  }
+
+  service_iam_role_name        = "${var.environment}-batch-app-batch-role"
+  service_iam_role_path        = "/batch/"
+  service_iam_role_description = "IAM service role for AWS Batch"
+  service_iam_role_tags = {
+    ModuleCreatedRole = "Yes"
+  }
+
+  create_spot_fleet_iam_role      = false
+  spot_fleet_iam_role_name        = "${var.environment}-batch-app-spot-role"
+  spot_fleet_iam_role_path        = "/batch/"
+  spot_fleet_iam_role_description = "IAM spot fleet role for AWS Batch"
+  spot_fleet_iam_role_tags = {
+    ModuleCreatedRole = "Yes"
+  }
+
+  compute_environments = {
+    a_fargate = {
+      name_prefix = "batch-app-fargate"
+
+      compute_resources = {
+        type      = "FARGATE"
+        max_vcpus = 4
+
+        security_group_ids = "${var.environment} =\"dev\" ? format(\"%s/%s\",\"DEV/APP\",upper(var.environment)) : upper(var.environment)}-OLCS-PRI-API-SG"
+        subnet_ids         = ["${var.environment} =\"dev\" ? format(\"%s/%s\",\"DEV/APP\",upper(var.environment)) : upper(var.environment)}-OLCS-PRI-BATCH-1A", "${var.environment} =\"dev\" ? format(\"%s/%s\",\"DEV/APP\",upper(var.environment)) : upper(var.environment)}-OLCS-PRI-BATCH-1B", "${var.environment} =\"dev\" ? format(\"%s/%s\",\"DEV/APP\",upper(var.environment)) : upper(var.environment)}-OLCS-PRI-BATCH-1C"]
+
+        # `tags = {}` here is not applicable for spot
+      }
+    }
+
+    b_fargate_spot = {
+      name_prefix = "batch-app-fargate_spot"
+
+      compute_resources = {
+        type      = "FARGATE_SPOT"
```

Comment:
Fargate spot isn't supported by ARM64 - https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-arm64.html. The images are ARM64.